router/batchrouter/batchrouter.go

```diff
@@ -15,37 +17,38 @@ import (
 	"github.com/rudderlabs/rudder-server/config"
 	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
 	"github.com/rudderlabs/rudder-server/jobsdb"
-	warehouseutils "github.com/rudderlabs/rudder-server/router/warehouse/utils"
 	"github.com/rudderlabs/rudder-server/rruntime"
 	"github.com/rudderlabs/rudder-server/services/filemanager"
 	"github.com/rudderlabs/rudder-server/services/stats"
 	"github.com/rudderlabs/rudder-server/utils"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 	"github.com/rudderlabs/rudder-server/utils/misc"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 	uuid "github.com/satori/go.uuid"
 	"github.com/tidwall/gjson"
 )
 
 var (
-	jobQueryBatchSize          int
-	noOfWorkers                int
-	maxFailedCountForJob       int
-	mainLoopSleep              time.Duration
-	uploadFreqInS              int64
-	configSubscriberLock       sync.RWMutex
-	objectStorageDestinations  []string
-	warehouseDestinations      []string
-	inProgressMap              map[string]bool
-	inProgressMapLock          sync.RWMutex
-	lastExecMap                map[string]int64
-	lastExecMapLock            sync.RWMutex
-	uploadedRawDataJobsCache   map[string]map[string]bool
-	warehouseStagingFilesTable string
+	jobQueryBatchSize         int
+	noOfWorkers               int
+	maxFailedCountForJob      int
+	mainLoopSleep             time.Duration
+	uploadFreqInS             int64
+	configSubscriberLock      sync.RWMutex
+	objectStorageDestinations []string
+	warehouseDestinations     []string
+	inProgressMap             map[string]bool
+	inProgressMapLock         sync.RWMutex
+	lastExecMap               map[string]int64
+	lastExecMapLock           sync.RWMutex
+	uploadedRawDataJobsCache  map[string]map[string]bool
+	warehouseURL              string
 )
 
 type HandleT struct {
 	destType          string
 	batchDestinations []DestinationT
+	netHandle         *http.Client
 	processQ          chan BatchJobsT
 	jobsDB            *jobsdb.HandleT
 	jobsDBHandle      *sql.DB
```

Comment:
In line 195 and 196, 
		opID = brt.jobsDB.JournalMarkStart(jobsdb.RawDataDestUploadOperation, opPayload)
		defer brt.jobsDB.JournalDeleteEntry(opID)

Because of defer, we always delete the journal entry irrespective of a crash - Then do we really need journaling here?

Reply:
changed the delete op to after we update job status.